amplify/data/resource.ts

```diff
@@ -80,18 +76,57 @@ const schema = a
         allow.group("Admin"),
         allow.authenticated().to(["read"]),
       ]),
-
     Team: a
       .model({
         name: a.string(),
         id: a.id(),
         approved: a.boolean(),
         members: a.hasMany("User", "teamId"),
+        scores: a.hasMany("Score", "teamId"),
+        teamRooms: a.hasMany("TeamRoom", "teamId"),
+      })
+      .authorization((allow) => [
+        allow.group("Admin").to(["read", "update"]),
+        allow.authenticated().to(["read"]),
+      ]),
+    Score: a
```

Comment:
Do we need to allow for scoring different aspects of a projects? Like multiple metrics of technical, pitch, etc., should we have a field for that (maybe u accounted for it somehow, idk lol)

Reply:
True actually

Since every team has a `scores: a.hasMany("Score", "teamId")`, why don't we add a field to `Score` like `scoreTitle` or something `scoreMetric` as mentioned where `scoreMetric: a.string().required`?

Reply:
I remember last year there were side pots for "Best use of AI", etc. So this way we could differentiate those scores to determine who wins the sidepots

Reply:
Yep very valid, we'll add the struct to a Hackathon metadata table and have it as the source of truth for the fields